### PR TITLE
Inspector support bounding non leaf nodes

### PR DESCRIPTION
**Pull request by TrevorDev**

### CODE REVIEW

File: tests/validation/config.json [hunk 1]
```diff
@@ -1,6 +1,12 @@
 ﻿{
   "root": "https://rawgit.com/BabylonJS/Website/master",
   "tests": [
+    {
+      "title": "Nested BBG",
+      "playgroundId": "#ZG0C8B#0",
+      "renderCount": 50,
```
Comment (bghgary) [inline tests/validation/config.json]: why do you need a renderCount?

File: inspector/src/components/sceneExplorer/entities/sceneTreeItemComponent.tsx [hunk 2]
```diff
@@ -206,6 +206,9 @@ export class SceneTreeItemComponent extends React.Component<ISceneTreeItemCompon
                     break;
                 case 4:
                     manager.boundingBoxGizmoEnabled = true;
+                    if(manager.gizmos.boundingBoxGizmo){
```
Comment (bghgary) [inline inspector/src/components/sceneExplorer/entities/sceneTreeItemComponent.tsx]:
```suggestion
                    if (manager.gizmos.boundingBoxGizmo) {
```

File: inspector/src/components/sceneExplorer/entities/sceneTreeItemComponent.tsx [hunk 3]
```diff
@@ -206,6 +206,9 @@ export class SceneTreeItemComponent extends React.Component<ISceneTreeItemCompon
                     break;
                 case 4:
                     manager.boundingBoxGizmoEnabled = true;
+                    if(manager.gizmos.boundingBoxGizmo){
+                        manager.gizmos.boundingBoxGizmo.fixedDragMeshScreenSize = true
```
Comment (bghgary) [inline inspector/src/components/sceneExplorer/entities/sceneTreeItemComponent.tsx]:
```suggestion
                        manager.gizmos.boundingBoxGizmo.fixedDragMeshScreenSize = true;
```

File: src/node.ts [hunk 4]
```diff
@@ -765,6 +766,13 @@ export class Node implements IBehaviorAware<Node> {
         this._behaviors = [];
     }
 
+    protected _getBoudingInfoMin() {
```
Comment (bghgary) [inline src/node.ts]: I don't understand why this is necessary.

Comment (TrevorDev) [reply src/node.ts]: I moved getHierarchyBoundingVectors to node. But it didn't contain certain things that abstractMesh has like .subMeshes which is why I added these functions, in abstractMesh it overrides them to get the same behavior it originally had when called from an abstract mesh.

File: src/node.ts [hunk 5]
```diff
@@ -779,4 +780,63 @@ export class Node implements IBehaviorAware<Node> {
...
+    public getHierarchyBoundingVectors(includeDescendants = true, predicate: Nullable<(abstractMesh: AbstractMesh) => boolean> = null): { min: Vector3, max: Vector3 } {
+        // Ensures that all world matrix will be recomputed.
+        this.getScene().incrementRenderId();
+
+        this.computeWorldMatrix(true);
+
+        let min: Vector3;
+        let max: Vector3;
+
+        let thisAbstractMesh = ((this as any) as AbstractMesh);
```
Comment (bghgary) [inline src/node.ts]:
```suggestion
        let thisAbstractMesh = (this as Node as AbstractMesh);
```

File: src/node.ts [hunk 6]
```diff
@@ -779,4 +780,63 @@ export class Node implements IBehaviorAware<Node> {
...
+        if (thisAbstractMesh.getBoundingInfo && thisAbstractMesh.subMeshes) {
+            // If this is an abstract mesh get its bounding info
+            let boundingInfo = thisAbstractMesh.getBoundingInfo();
+            min = boundingInfo.boundingBox.minimumWorld;
+            max = boundingInfo.boundingBox.maximumWorld;
+        }else {
```
Comment (bghgary) [inline src/node.ts]:
```suggestion
        } else {
```

File: src/node.ts [hunk 7]
```diff
@@ -779,4 +780,63 @@ export class Node implements IBehaviorAware<Node> {
             }
         }
     }
-}
\ No newline at end of file
+        /**
+     * Return the minimum and maximum world vectors of the entire hierarchy under current mesh
```
Comment (deltakosh) [inline src/node.ts]:
```suggestion
     * Return the minimum and maximum world vectors of the entire hierarchy under current node
```

**Pull request merged**